lib/keypair.rb

```diff
@@ -173,4 +206,25 @@ def set_keypair
     self._keypair ||= OpenSSL::PKey::RSA.new(2048).to_pem
     self.jwk_kid = public_jwk.kid
   end
+
+  # Set the validity timestamps based on the rotation interval.
+  def set_validity
+    self.not_before ||= created_at || Time.zone.now
+    self.not_after ||= not_before + ROTATION_INTERVAL
+    self.expires_at ||= not_after + ROTATION_INTERVAL
+  end
+
+  def not_after_after_not_before
+    return if not_before.nil? || not_after.nil?
+    return if not_after > not_before
+
+    errors.add(:not_after, 'must be after not before')
```

Comment:
If this would be an error that users could experience I would say that you need translations but this should in theory never happen so LGTM.